pkg/crypto/rsa.go

```diff
@@ -0,0 +1,53 @@
+package crypto
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io/ioutil"
+)
+
+const (
+	pubKeyIdentifier string = "PUBLIC KEY"
+)
+
+// LoadPublicKeyFromFile loads DER formatted RSA public key from file.
+func LoadPublicKeyFromFile(publicKeyPath string) (*rsa.PublicKey, error) {
+	x509PEM, err := ioutil.ReadFile(publicKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse x509 PEM file
+	block, _ := pem.Decode(x509PEM)
+	if block == nil || block.Type != pubKeyIdentifier {
+		return nil, errors.New("Can't decode PEM file")
+	}
+
+	// parse Public Key
+	pub, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	switch pub := pub.(type) {
+	case *rsa.PublicKey:
+		return pub, nil
+	default:
+		return nil, err
```

Comment:
This would return `nil, nil` as it's reusing `err` from the previous call, which is `nil`.

`return nil, errors.New("Not a public key")` or something like that should work

Reply:
got it